src/main/java/LunaBot.java

```diff
@@ -0,0 +1,245 @@
+        System.out.println(" What can I do for you?");
+        System.out.println("___________________________________________________________________");
+
+        // Create scanner for user input
+        Scanner scanner = new Scanner(System.in);
+        String input;
+
+        // loop scanner to echo user input
+        while (true) {
+            try {
+                input = scanner.nextLine();
+                if (input.equals("bye")) {
+                    System.out.println("___________________________________________________________________");
+                    System.out.println(" Bye. Hope to see you again soon!");
+                    System.out.println("___________________________________________________________________");
+                    break;
+                }
+                else if (input.equals("list")) {
+                    printTaskList();
+                }
+                else if (input.startsWith("mark ")) {
+                    markTask(input);
+                }
+                else if (input.startsWith("unmark ")) {
+                    unmarkTask(input);
+                }
+                else if (input.startsWith("delete ")) {
+                    deleteTask(input);
+                }
+                else if (input.startsWith("todo ")) {
+                    addTodo(input);
+                }
+                else if (input.startsWith("deadline ")) {
+                    addDeadline(input);
+                }
+                else if (input.startsWith("event ")) {
+                    addEvent(input);
+                }
+                else {
+                    throw new LunaBotException("Invalid command");
+                }
+            }
+            catch (LunaBotException e) {
+                System.out.println("___________________________________________________________________");
+                System.out.println(" Error: " + e.getMessage());
+                System.out.println("___________________________________________________________________");
+            }
+
+        }
+        scanner.close();
+    }
+
+    private static void printTaskList() {
+        System.out.println("___________________________________________________________________");
+        if (taskList.isEmpty()) {
+            System.out.println("Your task list is currently empty");
+        }
+        else {
+            System.out.println(" Here are the tasks in your list:");
+            for (int i = 0; i < taskList.size(); i++) {
+                System.out.println(" " + (i + 1) + ". " + taskList.get(i));
+            }
+        }
+        System.out.println("___________________________________________________________________");
+    }
+
+    // Method to mark tasks
+    private static void markTask(String input) throws LunaBotException {
+        try {
+            String taskNumString = input.substring(5);
+            // Checks to see if user provided a number
```

Comment:
I like how you add comments in the nested code to allow readers to know what you are doing at each stage of the code